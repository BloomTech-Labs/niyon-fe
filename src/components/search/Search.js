import React, { useContext, useState, useEffect } from 'react'
import { UserContext } from '../../UserContext'
import Header from '../header/Header'
import Footer from '../footer/Footer'
import Select from 'react-select'
import { job } from '../profile/job'
import { location } from '../profile/location'
import { technology } from '../profile/technologies'
import { axiosWithAuth } from '../apiStuff/axiosWithAuth'
import SwipeTabsJobTitle from './SwipeTabsJobTitle'
import SwipeTabsLocation from './SwipeTabsLocation'
import SwipeTabsTechnology from './SwipeTabsTechnology'

export function Search (props) {
  /*eslint-disable */
  const { user, setUser } = useContext(UserContext)
  const [profiles, setProfiles] = useState([])
  const [jobTitlesToDisplay, setJobTitlesToDisplay] = useState([])
  const [locationsToDisplay, setLocationsToDisplay] = useState([])
  const [technologiesToDisplay, setTechnologiesToDisplay] = useState([])
  const jobs = job
  const locations = location
  const technologies = technology

  const handleJobChange = (selectedItem) => {
    if (selectedItem) {
      const usersToDisplayByJobTitle = profiles.filter(
        (job) => job.job_title_id === selectedItem.value
      )
      setJobTitlesToDisplay(usersToDisplayByJobTitle)
    }
  }

  const handleLocationChange = (selectedItem) => {
    if (selectedItem) {
      const usersToDisplayByLocation = profiles.filter(
        (location) => location.location_id === selectedItem.value
      )
      setLocationsToDisplay(usersToDisplayByLocation)
    }
  }

  const handleTechChange = (selectedItem) => {
    if (selectedItem) {      
      const selectedTechStack = selectedItem.map(item => item.value)
      const usersWhoHaveSelectedTechs = []
      const mappingFunction = profiles.map(profile => {
        if (selectedTechStack.every(value => profile.techs.includes(value))) {
          usersWhoHaveSelectedTechs.push(profile)
          setTechnologiesToDisplay(usersWhoHaveSelectedTechs)
        }
      })     
    }
  }

  useEffect(() => {
    const apiCall = async () => {
      await axiosWithAuth()
        .get('/profile')
        .then((res) => {
          if (res) {
            const data = res.data
            setProfiles(data)
          }
        })
        .catch((err) => console.log(err))
    }
    apiCall()
  }, [])

  return (
    <div>
      <Header />
      <div className="search" data-test="search-container">
        <div>
          <h1 className="search-header">Search</h1>
          <h2 data-test="job-title">Job Title</h2>
          <Select
            name="job_title_id"
            options={jobs}
            className="basic-multi-select"
            classNamePrefix="select"
            onChange={handleJobChange}
            data-test="job-title-search"
          />
          <p>Users with Selected Job Title</p>
          <SwipeTabsJobTitle jobTitlesToDisplay={jobTitlesToDisplay} />
          {/* {jobTitlesToDisplay.map((profile) => (
            <div key={profile.id}>
              {profile.first_name}
              {profile.last_name}
              {profile.user_type}
              {profile.location}
            </div>
          ))} */}
          <h2>Location</h2>
          <Select
            name="location_id"
            options={locations}
            className="basic-multi-select"
            classNamePrefix="select"
            onChange={handleLocationChange}
          />
          <p>Users with Selected Location</p>
          <SwipeTabsLocation locationsToDisplay={locationsToDisplay} />
          {/* {locationsToDisplay.map((profile) => (
            <div key={profile.id}>
              {profile.first_name}
              {profile.last_name}
              {profile.user_type}
              {profile.location}
            </div>
          ))} */}
          <h2>Technologies</h2>
          <Select
            isMulti
            name="techs"
            options={technologies}
            className="basic-multi-select"
            classNamePrefix="select"
            onChange={handleTechChange}
          />
          <p>Users with Selected Technology</p>
          <SwipeTabsTechnology technologiesToDisplay={technologiesToDisplay} />
          {/* {technologiesToDisplay.map((profile) => (
            <div key={profile.id}>
              {profile.first_name}
              {profile.last_name}
              {profile.user_type}
              {profile.location}
            </div>
          ))} */}
        </div>
      </div>
      <Footer value={2} />
    </div>
  )
}

export default Search
