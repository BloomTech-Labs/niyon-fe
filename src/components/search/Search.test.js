import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Search from './Search';
import { MemoryRouter } from 'react-router-dom';
import findByTestAttr from '../../tests/utils';
import  { UserContext } from '../../UserContext';

// const setUp = (props={}) => {
//    const wrapper = shallow(<Search />);
//    return wrapper;
// }

describe('<Search /> component testing', () => {
   const user = {}  ;
   const setUser = jest.fn();
   let component;
   const profiles = [];
   beforeEach(() => {
      component = mount(
          <UserContext.Provider value={{user,setUser}}>
           <MemoryRouter initialEntries={['/search']}>
               <Search />
               </MemoryRouter>    
          </UserContext.Provider>
      )
   });

//    it('should pass snaps shot testing', () => {
//         expect(toJSON(component)).toMatchSnapshot();
//    });

   it('should render the <Search /> component correctly', () => { 
        expect(component.exists()).toBe(true);
   });
   it('should render <Header /> component correctly', () => {
        expect(component.find('Header').length).toBe(1);
   });

   it('should render container wrapper correctly', () => {
         const divContainer = findByTestAttr(component, 'search-container');
         expect(divContainer.length).toBe(1);
   });

   it('should render main title correctly', () => {
        expect(component.find('h1').exists()).toBe(true);
        expect(component.find('h1').render().text().trim()).toEqual('Search');
   });

   it('should render search title correctly', () => {
     expect(component.find('h2').exists()).toBe(true);
     expect(component.find('h2').render().text().trim()).toEqual('Job Title');
   });

   describe('<select /> component', () => {
         let selectWrapper;
         const setState = jest.fn();
         const useStateSpy = jest.spyOn(React, 'useState');
         useStateSpy.mockImplementation((init) => [init, setState]);
         const handleChange = setState();
         beforeEach(() => {
              selectWrapper = component.find('Select');
         });

         it('should render <Select />  component correctly', () => {
            expect(selectWrapper.exists()).toBe(true);
         
        });
        it('it should change the job title on change correctly', () => {
            selectWrapper.props().onChange(handleChange);
            expect(setState).toHaveBeenCalled();
        });
   });

   it('should render the selected job title paragraph correctly', () => {
     expect(component.find('p').length).toBe(1);
     expect(component.find('p')
                     .render()
                     .text()
                     .trim())
                     .toEqual('Users with Selected Job Title');
    });

    it('should render profiles container correctly', () => {
          const container = findByTestAttr(component, 'search-profile');
          if(profiles.length === 0) {
            expect(container.length).toBe(0);
          } else {
            expect(container.length).toBe(1);
          }
    })

   it('should render the <Footer /> component correctly', () => {
        expect(component.find('Footer').length).toBe(1);
   });
});