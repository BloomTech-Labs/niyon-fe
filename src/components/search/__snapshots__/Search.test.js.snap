// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component testing should pass snaps shot testing 1`] = `
<div>
  <Header />
  <div
    className="search"
    data-test="search-container"
  >
    <div>
      <h1
        className="search-header"
      >
        Search
      </h1>
    </div>
  </div>
  <Footer
    value={2}
  />
</div>
`;
